src/components/create-safe/status/hooks/useWatchSafeCreation.ts

```diff
@@ -35,8 +38,26 @@ const useWatchSafeCreation = ({
 
     if (status === SafeCreationStatus.INDEXED) {
       trackEvent(CREATE_SAFE_EVENTS.GET_STARTED)
+      const chainPrefix = chain?.shortName
 
-      safeAddress && router.push({ pathname: AppRoutes.safe.home, query: { safe: safeAddress } })
+      if (safeAddress && chainPrefix) {
+        const address = `${chainPrefix}:${safeAddress}`
+        const redirectUrl = router.query?.safeViewRedirectURL
+        if (typeof redirectUrl === 'string') {
+          // We're prepending the safe address directly here because the `router.push` doesn't parse
+          // The URL for already existing query params
+          const hasQueryParams = redirectUrl.includes('?')
+          const appendChar = hasQueryParams ? '&' : '?'
+
+          if (redirectUrl.includes('apps')) {
+            trackEvent({ ...SAFE_APPS_EVENTS.SHARED_APP_OPEN_AFTER_SAFE_CREATION })
```

Comment:
Can we track this on the share page instead?

Reply:
How? We don't go back to the share page after the safe creation

Reply:
To the safe + app page directly